tf2_ros_py/tf2_ros/buffer.py

```diff
@@ -87,13 +88,27 @@ def __init__(
         if node is not None:
             self.srv = node.create_service(FrameGraph, 'tf2_frames', self.__get_frames)
 
+        # create a jump callback so as to clear the buffer if use_sim_true is true and there is a jump in time
+        threshold = JumpThreshold(min_forward=Duration(seconds=0),
+                                  min_backward=Duration(seconds=-1),
+                                  on_clock_change=True)
+        self.jump_handle = rclpy.clock.Clock().create_jump_callback
+        (
+            threshold,
+            post_callback=self.time_jump_callback
+        )
```

Comment:
I think this needs to be done in a slightly different way.

In particular, I'm pretty sure that the clock object has to be kept around, otherwise this whole thing will be destroyed immediately.

There are then two paths we have to deal with; when the node is passed in, and when it is not.  When it is passed in, we can just use the clock object that is stored in the Node.  When it is not, then we have to create and hold onto a clock object ourselves.  So something like:

```
if node is not None:
    self.jump_handle = node.get_clock().create_jump_callback(threshold, post_callback=self.time_jump_callback)
else:
    self.clock = rclpy.clock.Clock()
    self.jump_handle = self.clock.create_jump_callback(threshold, post_callback=self.time_jump_callback)
```

(that `else` case is actually problematic since it isn't respecting `use_sim_time` for simulation; the only way to deal with that would be to make `node` non-optional here.  But we can do that later.)